tests/lib/receivers.js

```diff
@@ -288,4 +289,43 @@ describe('Receivers', function () {
       xhr.abort();
     });
   });
+  describe('info', function () {
+    it('will timeout - default timeout', function (done) {
+      this.timeout(10000);
+      InfoReceiver.prototype._getReceiver = function(baseUrl, url) {
+        return new InfoAjax(url, XhrFake);
+      };
+
+      var expectedWasClean = true;
+      InfoReceiver.prototype._cleanup = function(wasClean) {
+        expect(wasClean).to.equal(expectedWasClean);
+        if (expectedWasClean === false) {
+          // Cleanup was called because of a timeout
+          done();
+        }
+        expectedWasClean = false;
+      };
+
+      new InfoReceiver('test', {});
+    });
+
+    it('will timeout - configured timeout', function (done) {
+      this.timeout(2000);
+      InfoReceiver.prototype._getReceiver = function(baseUrl, url) {
+        return new InfoAjax(url, XhrFake);
+      };
+
+      var expectedWasClean = true;
+      InfoReceiver.prototype._cleanup = function(wasClean) {
+        expect(wasClean).to.equal(expectedWasClean);
+        if (expectedWasClean === false) {
+          // Cleanup was called because of a timeout
+          done();
+        }
+        expectedWasClean = false;
+      };
+
+      new InfoReceiver('test', {}, 1000);
+    });
+  });
```

Comment:
I apologize for these awkward tests - I do not fully understand all of the inheritance going on, so I overrode what I needed to test the override.  The result is that the finish event is still emitted, and that is why I have the `wasClean` expectation change.